### PR TITLE
Intersect edges performance

### PR DESCRIPTION
**Pull request by portnov**
## Addressed problem description

"Intersect edges MK2"  node had pretty bad performance. This was discussed in #1990.

## Solution description

* Simplify addressing vertices in the BMesh
* Do not check for edges intersection if their bounding boxes do not intersect
* Do not check for edges intersection if they are not coplanar.

On my test scenes, for relatively large meshes, this reduced calculation time in about 3-4 times.

## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [x] Unit-tests implemented.
- [x] Ready for merge.

@zeffii Please review.

### CODE REVIEW

**Comment by zeffii** [comment]
i'll read the code now

File: utils/cad_module_class.py [hunk 1]
```diff
@@ -136,9 +155,21 @@ def vertex_indices_from_edges_tuple(self, bm, edge_tuple):
         > edge_tuple:   contains two edge indices.
         < returns the vertex indices of edge_tuple
         '''
-        k = lambda v, w: bm.edges[edge_tuple[v]].verts[w].index
-        return [k(i >> 1, i % 2) for i in range(4)]
+        e1, e2 = edge_tuple
+        bm_e1, bm_e2 = bm.edges[e1], bm.edges[e2]
+        return [bm_e1.verts[0].index, bm_e1.verts[1].index,
+                bm_e2.verts[0].index, bm_e2.verts[1].index]
```
Comment (zeffii) [inline utils/cad_module_class.py]: i don't know what possessed me to use bitwise operators and a lambda for this :)

**Comment by zeffii** [comment]
looks good to me! go go go!

**Comment by portnov** [comment]
Okay :)

**Pull request merged**